setup/tue.bash_functions

```diff
@@ -722,23 +851,79 @@ function tue-robocup-set-timezone-home
     sudo timedatectl set-timezone Europe/Amsterdam
 }
 
+function _ping_bool
+{
+    ping -c 1 "$1" 1>/dev/null 2>/dev/null
+    if [ "$?" == "0" ]
+    then
+        return 0
+    else
+        return 1
+    fi
+}
+
 function tue-robocup-install-package
 {
-    local pkgs_dir=$TUE_ENV_DIR/repos/https_/github.com/tue-robotics
-    # replace spaces with underscores
-    pkgs_dir=${pkgs_dir// /_}
-    # now, clean out anything that's not alphanumeric or an underscore
-    pkgs_dir=${pkgs_dir//[^a-zA-Z0-9\/\.-]/_}
+    local repos_dir=$TUE_ENV_DIR/repos/https_/github.com/tue-robotics
+    local repo_dir=$repos_dir/${1}.git
 
-    local pkg_dir=$pkgs_dir/${1}.git
+    local mem_pwd=$PWD
 
+    local remote="roboticssrv"
+    local server="amigo@roboticssrv.local:"
+    local branch="robocuo"
 
     # If directory already exists, return
-    [ -d $pkg_dir ] && return
+    [ -d $repo_dir ] && return
+
+    # If there is internet, use it for first clone. Otherwise an error
+    # will be raised during switching back to default branch of origin
+    local internet=$(echo _ping_bool github.com)
+    if $internet
+    then
+        git clone https://github.com/tue-robotics/${1}.git $repo_dir
```

Comment:
I don't like different behavior based on whether or not internet is available. Can you make it that internet is required? You can always do a clone in the hotel and most of the time internet is available.

Reply:
The behind this function (in my mind) is:
installing a package without internet available and also setting it up correctly, so that is works correctly with all the other tools.

If you have internet you could just use tue-get, but than it is not setup to use the server.  The current tool to switch to the server, shouldn't be used during a tournament. Because it checks out the robocup branch for all repos, which is bad if you are actually on an other branch at that moment.

I could change this to only checking out robocup, if you are on a branch which has github as a remote. Then you could run the tool to switch to the server also during tournament.

Then the tue-robocup-install-package is only for situations without internet. This has only one issue with the tool for switching back to github. That tool tries to checkout the symbolic ref `refs/remotes/origin/HEAD`. This is only created for the remote used at cloning. So if you clone from the server, rename that one to roboticssrv. Then add github as origin, the symbolic ref doesn't exist for origin. (This is also a problem right now and the reason for the different behaviour, because you would like to avoid that situation.). If I change the function that it will checkout master if no symbolic ref exists, that is fixed.

For with internet you then could you use:
tue-get install TARGET
tue-robocup-set-server

For without internet:
tue-robocup-install-package PKG